openlibrary/templates/search/lists.html

```diff
@@ -1,30 +1,53 @@
-$def with (q, lists)
+$def with (get_results)
+
+$ q = query_param('q', '').strip()
+$ results_per_page = 10
+$ page = query_param('page')
+$if page:
+    $ page = int(page)
+$else:
+    $ page = 1
+$ offset = (page - 1) * results_per_page
 
 $var title: $_('Search results for Lists')
 
 <div id="contentHead">
-  <h1>$_("Search Lists")</h1>
+    <h1>$_("Search Lists")</h1>
 </div>
 
 <div id="contentBody">
-  $:macros.SearchNavigation()
-  <div class="section">
-    <form class="siteSearch olform" action="/search/lists">
-      $:render_template("search/searchbox", q=q)
-    </form>
-  </div>
-
-  $if lists:
-    <div class="mybooks-list">
-      <ul id="listResults" class="list-books">
-      $for list in lists:
-        <li class="searchResultItem" id="list-$loop.index">$:render_template("lists/preview", list)</li>
-      </ul>
-    </div>
-  $elif q:
-    <div>
-      <span class="red">$_("No results found.")</span>
-      <a href="/search/inside?$urlencode(dict(q=q))">$_('Search for books containing the phrase "%s"?', q)</a>
+    $:macros.SearchNavigation()
+
+    <div class="section">
+        <form class="siteSearch olform" action="">
+            $:render_template("search/searchbox", q=q)
+        </form>
     </div>
 
-</div>
+    <div id="contentMeta">
+        $ results = get_results(q, offset=offset, limit=results_per_page)
+
+        $if q and results.error:
+            <strong>
+                $for line in results.error.splitlines():
+                    $line
+                    $if not loop.last:
+                        <br>
+            </strong>
+
+        $if q and not results.error:
+            <div class="mybooks-list">
+                $if results.num_found:
+                    <ul id="listResults" class="list-books">
+                        $for list in lists:
```

Comment:
I think I was suppose to change this line into
`$for list in results:`
because the original `lists` in `$def with (q, lists)` is no more.

Lastly, what do you think about the "preview.html" below this line? Any changes required?